src/backend/aspen/api/views/auth.py

```diff
@@ -20,17 +25,66 @@
 @router.get("/login")
 async def login(
     request: Request,
+    organization: Optional[str] = None,
+    invitation: Optional[str] = None,
+    organization_name: Optional[str] = None,
     auth0: StarletteOAuth2App = Depends(get_auth0_client),
     settings: Settings = Depends(get_settings),
 ) -> Response:
-    return await auth0.authorize_redirect(request, settings.AUTH0_CALLBACK_URL)
+    kwargs = {}
+    if invitation:
+        kwargs["invitation"] = invitation
+    if organization:
+        kwargs["organization"] = organization
+    if organization_name:
+        kwargs["organization_name"] = organization_name
+    return await auth0.authorize_redirect(
+        request, settings.AUTH0_CALLBACK_URL, **kwargs
+    )
+
+
+async def create_user_if_not_exists(db, userinfo):
+    if "org_id" not in userinfo:
+        return  # We're currently only creating new users if they're confirming an org invitation
```

Comment:
Feels like it should be an exception getting raised here

Reply:
Our main Auth0 application supports two login modes: 
1. Logging into our app without an organization. This is what we do right now, and will still be the default login mode for CZGE
2. Logging into our app *with* an organization. This is the flow that gets used when accepting an org invitation, and this flow can be forced by adding a query string parameter to our login URL.

We want to continue supporting both flows (at least for now) so not all auth tokens will contain an org ID.

Reply:
ahhh, got it, thank you. I thought (2) was totally supplanting (1), that makes sense